feat(issues): add subtitles for common search filters

shayna-ch opened the pull request:
Added descriptions to the `is:` field value suggestions. References #83147

`IsFieldDescriptions`, `FIELD_VALUE_TO_FIELD_DESCRIPTION`, and `getIsFieldDescriptionFromValue` are copied from how subtitles are created for `isue.type`. 

- `resolved` → "Issues marked as fixed"
- `unresolved` → "Issues still active and needing attention"
- `archived` → "Issues marked as archived"
- `escalating` → "Issues occurring significantly more often than they used to"
- `new` → "Issues that first occurred in the last 7 days"
- `ongoing` → "Issues created more than 7 days ago or manually been marked as reviewed"
- `regressed` → "Issues resolved then occurred again"
- `assigned` → "Issues assigned to a team member"
- `unassigned` → "Issues not assigned to anyone"
- `for_review` → "Issues pending review"
- `linked` → "Issues linked to other issues"
- `unlinked` → "Issues not linked to other issues"

<img width="276" height="394" alt="Screenshot 2025-07-24 at 3 22 19 PM" src="https://github.com/user-attachments/assets/cdb29988-659f-4f51-9f65-ac81aa24a3bf" />

shayna-ch committed:
feat(issues): add subtitles for common search filters

shayna-ch committed:
feat(issues): add subtitles for common search filters

shayna-ch committed:
feat(issues): add subtitles for common search filters

shayna-ch committed:
feat(issues): add subtitles for common search filters

shayna-ch committed:
feat(issues): add subtitles for common search filters

fix export unused failure

thetruecpaul commented on static/app/utils/fields/index.ts:
```diff
@@ -403,6 +403,46 @@ export enum IsFieldValues {
+};
+
+export function getIsFieldDescriptionFromValue(
+  isFieldValue: string
```
Can we tighten this type to `IsFieldValues`?

thetruecpaul commented on static/app/utils/fields/index.ts:
```diff
@@ -403,6 +403,46 @@ export enum IsFieldValues {
+  isFieldValue: string
+): IsFieldDescriptions | undefined {
+  if (isFieldValue in FIELD_VALUE_TO_FIELD_DESCRIPTION) {
+    // @ts-expect-error TS(7053): Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
```
Can we remove the type error if we tinker around here (maybe by changing the keys of `FIELD_VALUE_TO_FIELD_DESCRIPTION` to explicity use `IsFieldValues` & updating the input type)? Not blocking if we can't, but would prefer to avoid introducing new type errors.

scttcper commented on static/app/utils/fields/index.ts:
```diff
@@ -403,6 +403,46 @@ export enum IsFieldValues {
 }
 
+enum IsFieldDescriptions {
+  RESOLVED = 'Issues marked as fixed',
```
strings should be translated with our t function
```suggestion
  RESOLVED = t('Issues marked as fixed'),
```

shayna-ch committed:
feat(issues): add subtitles for common search filters

- changed IsFieldDescriptions to an object
- added translations with t function
- changed isFieldValue type to IsFieldValues which fixed ts error

shayna-ch committed:
feat(issues): add subtitles for common search filters

- change min-width back to 200px

malwilley commented on static/app/utils/fields/index.ts:
```diff
@@ -403,6 +403,45 @@ export enum IsFieldValues {
+  UNLINKED: t('Issues not linked to other issues'),
+};
+
+const FIELD_VALUE_TO_FIELD_DESCRIPTION = {
```
nit: I would type this as `Record<IsFieldValue, string>` and just do

```ts
[IsFieldValue.RESOVLED]: t('Issues marked as fixed'),
// ...etc
```

That way if any new values get added, this type will error until a new string is added

shayna-ch committed:
feat(issues): add subtitles for common search filters

- changed type to Record<IsFieldValues, string>
